src/ZendViewRenderer.php

```diff
@@ -342,7 +342,7 @@ private function prepareLayout(ModelInterface $viewModel)
         if ($layout) {
             $layout->addChild($viewModel);
             $viewModel = $layout;
-            $viewModel->setVariables($this->mergeParams(self::TEMPLATE_ALL, []));
+            $viewModel->setVariables($this->mergeParams($layout->getTemplate(), []));
```

Comment:
Note: TEMPLATE_ALL is applied implicitly in trait. With this change merged parameters in effect look like all default + layout default + [].

I see a problem here, in case of $providedLayout, this will discard variables in layout view model. Proper should be 
```php
$viewModel->setVariables($this->mergeParams($layout->getTemplate(), $layout->getVariables()));
```

@dennis-fedco can you please also add test that asserts that layout variable provided this way is available and not overriden by default.
```php
$layout = new ViewModel(['title' => $title]);
$layout->setTemplate('zendview-layout-variable');
$renderer->render(..., ['layout' => new ViewModel()]);
```
 And another test for layout provided this way that default is still applied.

Reply:
When I add `$layout` the way you have in the last 3 lines of code, all the tests still path as before, without any changes to the code. 

In the new test I removed the line with `addDefaultParam` and I removed passing the layout parameter to `ZendViewRenderer`'s constructor, and I pass the `$layout` to the `render` method as you've mentioned.  So I must not be triggering the requested codepath for provided layout.

I'll look into it a little bit, but is there another way to trigger the behavior you are describing?

Reply:
Pretty much same test as yours, except default parameter 'title' should receive different value that is not expected to appear